pkg/url/url.go

```diff
@@ -904,15 +959,42 @@ func Push(client *occlient.Client, kClient *kclient.Client, parameters PushParam
 				if endpoint.Path != "" {
 					path = endpoint.Path
 				}
-				urlLOCAL[url.Name] = URL{
-					Spec: URLSpec{
-						Host:      url.Host,
-						Port:      url.Port,
-						Secure:    secure,
-						TLSSecret: url.TLSSecret,
-						Kind:      url.Kind,
-						Path:      path,
-					},
+				name := strings.TrimSpace(util.GetDNS1123Name(strings.ToLower(endpoint.Name)))
+				name = util.TruncateString(name, 15)
+				envUrls := parameters.EnvURLS
+				existInEnv := false
+				for _, url := range envUrls {
```

Comment:
put these in a map outside the loops and fetch from it

Reply:
Why does it matter? Convert a slice into a map needs to iterating through the entire slice. 
We can break the loop once we find the matching name when looping through the slice inside.

Reply:
same as https://github.com/openshift/odo/pull/3857#discussion_r483126733

Reply:
updated